core/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -415,6 +415,12 @@ public XContentBuilder toInnerXContent(XContentBuilder builder, Params params) t
             builder.field(Fields._NODE, shard.getNodeIdText());
         }
         if (nestedIdentity != null) {
+            if (index != null) {
+                builder.field(Fields._INDEX, RemoteClusterAware.buildRemoteIndexName(clusterAlias, index));
+            }
+            if (id != null) {
+                builder.field(Fields._ID, id);
+            }
             nestedIdentity.toXContent(builder, params);
         } else {
             if (index != null) {
```

Comment:
Looks like the index and id rendering could be pulled out regardless of whether nestedIdentity is set or not now. Also wondering why we render type information in the "else" branch but not for the nestedIdentity case here, should this still be there in both cases until types are removed completely?

Reply:
@cbuescher Good point. I'll also include _type for nested hits. So that this code becomes cleaner and the normal hits and nested hits become more consistent.